File: 1029.py
def precompute_fib(max_n):
    fib = [0] * (max_n + 2)
    fib[0] = 0
    fib[1] = 1
    for i in range(2, max_n + 2):
        fib[i] = fib[i - 1] + fib[i - 2]
    return fib

cases = int(input())
inputs = [int(input()) for _ in range(cases)]
max_n = max(inputs)

fib = precompute_fib(max_n + 1)

for num in inputs:
    calls = fib[num]
    total_calls = 2 * fib[num + 1] - 2
    print(f"fib({num}) = {total_calls} calls = {calls}")

File: 1029.c
#include <stdio.h>

int fibonacci(int num, int *count, int *calls) {
    (*count)++;

    if(num == 0) {
        return 0;
    }else if(num == 1) {
        (*calls)++;
        return 1;
    }else {
        return fibonacci(num - 1, count, calls) + fibonacci(num - 2, count, calls);
    }
}

int main() {
    int cases, num, i;
    int count, calls;

    scanf("%d", &cases);

    for(i = 0; i < cases; i++) {
        count = -1;
        calls = 0;
        scanf("%d", &num);
        fibonacci(num, &count, &calls);
        printf("fib(%d) = %d calls = %d\n", num, count, calls);
    }

    return 0;
}